### ISSUE TITLE
 Error: failed to download "jupyterhub/binderhub" 

### ISSUE THREAD

**Issue by ctr26** [Praqma/helmsman#512]
Hi all,

I'm trying to incorporate helmsman into my CI/CD for binderhub, unfortunately even a bare bones config fails "While getting chart information"

Looking at the debug I think the issue is that helmsman runs `helm show chart jupyterhub/binderhub` without a version number and for binderhub that returns nothing:

`helm show chart jupyterhub/binderhub --version 0.2.0-n219.hbc17443`

```
apiVersion: v1
dependencies:
- name: jupyterhub
  repository: https://jupyterhub.github.io/helm-chart
  version: 0.9.0
description: A helm chart to install Binder
name: binderhub
version: 0.2.0-n219.hbc17443
```

`helm show chart jupyterhub/binderhub`

`Error: failed to download "jupyterhub/binderhub" (hint: running `helm repo update` may help)`

Here's my config:

```
namespaces:
  binderhub-staging:

helmRepos:
  jupyterhub: "https://jupyterhub.github.io/helm-chart/"

apps:
  binderhub-staging:
    valuesFiles:
      - "binderhub/config.yaml"
      - "binderhub/secret.yaml"
    name: "binderhub-staging"
    namespace: "binderhub-staging"
    chart: "jupyterhub/binderhub"
    version: "0.2.0-n219.hbc17443"
    priority: -3

```

**Comment by mkubaczyk** [comment]
If the version of a chart is not being passed to helm show chart command, it seeks for latest one. Somehow your helm repo does not let helm know what the latest is. How is your repo deployed? I checked that on private repo of mine and there's everything fine event though I didn't have to explicitly mark anything as latest, it just takes the latest one date-wise I suppose.

**Comment by ctr26** [comment]
Binderhub (not my chart) doesnt install unless you specify a version latest. I'm assuming that's because it's helm3 specific and helm3 requires a version number.BestCraigOn 18 Sep 2020 08:11, Mateusz Kubaczyk <notifications@github.com> wrote:
If the version of a chart is not being passed to helm show chart command, it seeks for latest one. Somehow your helm repo does not let helm know what the latest is. How is your repo deployed? I checked that on private repo of mine and there's everything fine event though I didn't have to explicitly mark anything as latest, it just takes the latest one date-wise I suppose.

**Comment by ctr26** [comment]
N.b. that  jupyterhub/jupyterhub (where jupyterhub/binderhub doesn't) pulls the latest version fine, so it's definitely how they've set their helm repo up.On 18 Sep 2020 08:28, ctr26@ebi.ac.uk wrote:Binderhub (not my chart) doesnt install unless you specify a version latest. I'm assuming that's because it's helm3 specific and helm3 requires a version number.BestCraigOn 18 Sep 2020 08:11, Mateusz Kubaczyk <notifications@github.com> wrote:
If the version of a chart is not being passed to helm show chart command, it seeks for latest one. Somehow your helm repo does not let helm know what the latest is. How is your repo deployed? I checked that on private repo of mine and there's everything fine event though I didn't have to explicitly mark anything as latest, it just takes the latest one date-wise I suppose.

**Comment by ctr26** [comment]
Any news on this I can't use helmsman in my CI until this is addressed

**Comment by mkubaczyk** [comment]
I could not understand what you've written here. Can you please answer once again to what I typed before? ;-) Just please be careful with the formatting.

**Comment by ctr26** [comment]
Looks like my text got garbled responding via email.

`Binderhub (not my chart) doesnt install unless you specify a version latest. I'm assuming that's because it's helm3 specific and helm3 requires a version number.`

'N.b. that  jupyterhub/jupyterhub (where jupyterhub/binderhub doesn't) pulls the latest version fine, so it's definitely how they've set their helm repo up'

TL;Dr the helm repo I'm using (jupyterhub/binderhub) is not mine; it is setup in such a way that it it doesn't return the latest version, so you have to specify a version. But, helmsman makes a call to 

`helm show chart jupyterhub/binderhub`

As part of it's setup process to get metadata (I think), which causes an error (shown above). Whereas including the version number in the setup

`helm show chart jupyterhub/binderhub --version 0.2.0-n219.hbc17443`

As an example, would work; but helmsman doesn't do this.

**Comment by antoinedeschenes** [comment]
I've had a similar problem 2 weeks ago when the latest release of another helm repository was not published correctly. `helm show chart` with `--version` still worked properly however. I suppose it would be safer to check the chart name in the known-good version than assuming latest has the correct information.

**Comment by antoinedeschenes** [comment]
In the case of jupyterhub/binderhub, helm code looks for the first matching release, excluding prereleases https://github.com/helm/helm/blob/fc9b46067f8f24a90b52eba31e09b31e69011e93/pkg/repo/index.go#L158. (That's the behavior of constraint '*')

If one looks at https://jupyterhub.github.io/helm-chart/index.yaml, there are only prereleases in the chart index.

**Comment by ctr26** [comment]
> I've had a similar problem 2 weeks ago when the latest release of another helm repository was not published correctly. `helm show chart` with `--version` still worked properly however. I suppose it would be safer to check the chart name in the known-good version than assuming latest has the correct information.

Helm show chart with the version should be how this handled, I agree.

**Comment by ndegory** [comment]
This issue looks like #530, for which there's a simple solution implemented in #532

**Comment by ctr26** [comment]
#532 is more elegant than #528, well done!

**Comment by ctr26** [comment]
Running binderhub with latest version of  helmsman using --devel still doesn't work unfortunately. My patch at #528 does but I don't know why it does vs #530
